enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,511 @@
+for details.
+
+#### machine-config-operator
+
+The machine-config-operator includes a check in code that the number
+of control plane nodes is 3. This code needs to be updated to look at
+the new high-availability mode capability API to determine the number
+of expected nodes to allow the operator to complete its work, for
+example to enable the realtime kernel on a single node with the
+appropriate performance profile settings.
+
+The machine-config-operator tries to drain a node before applying
+local changes and rebooting. In a true single-node deployment, there
+is nowhere for drained pods to be rescheduled. In a deployment with a
+single-node control plane and additional workers, some nodes could
+potentially be drained, but there is no way for the MCO to know if the
+workers are fully utilized. When the high-availability mode is `None`
+the step needs to be skipped. Ideally we will be able to rely on the
+[graceful shutdown support in
+kubelet](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown)
+to stop all workloads safely as part of the reboot.
+
+#### cluster-authentication-operator
+
+By default, the `cluster-authentication-operator` will not deploy
+oauth-apiserver and oauth-server without a minimum of 3 control plane
+nodes. This can be change by enabling
+`useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`. The
+operator will need to be updated to honor the new capabilities API
+instead. Health checks for these components will also have to be
+updated to allow a single replica.
+
+#### cluster-etcd-operator
+
+By default, `cluster-etcd-operator` will not deploy the etcd cluster
+without minimum of 3 master nodes. This can be changed by enabling
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd`. The operator
+will need to be updated to honor the new capabilities API instead.
+
+`etcd-quorum-guard` is managed by the cluster-version-operator, which
+means it has a statically configured Deployment. The operator will
+need to be updated to manage the `etcd-quorum-guard` so that it can
+support changing the replica count or disabling the guard entirely,
+depending on the best approach to honor the high-availability mode
+setting in the capabilities API.
+
+#### cluster-ingress-operator
+
+By default, the `cluster-ingress-operator` deploys the router with 2
+replicas. On a single node one will fail to start and the ingress will
+show as degraded. The operator will need to be updated to honor the
+new capabilities API to change the replica count to 1 when the
+high-availability mode is none.
+
+#### cluster-machine-approver
+
+The operator should disable automatic machine approval. (See risks
+section below.)
+
+#### operator-lifecycle-manager
+
+Operators managed by the `operator-lifecycle-manager` will look at the
+`infrastructure` API if they need to configure their operand
+differently based on the high-availability mode.
+
+### Risks and Mitigations
+
```

Comment:
what does DR look like? does it exist? to what extent?

Reply:
The "rebuild it" approach is described a bit in the two sections on lines 318-331. Do those address your concerns?